higher-order.js

```diff
@@ -95,6 +95,16 @@ const _ = {
         }, []);
     },
 
+    /**
+     * Given an array and filter, returns a new array with elements of the original array that don't apply to the rejection closure.
+     * @param {Array} arr - The array to be filtered.
+     * @param {Function} rejectionApplier - The function in charge of deciding if a value should be included in the filtered array.
+     * @returns {Array} filteredArray - The filtered array.
+     */
+    reject: function(arr, rejectionApplier) {
```

Comment:
I probably wouldn't give each function argument a different name related to the higher-order function that it's being passed to, e.g., `reject` and `rejectionApplier`, `filter` and `filterApplier`, etc.

I'd give them a generic name like `fn`, `func`, `callback`, etc.

Normally I'd say your approach is right — name a thing after its role in the code — but because this is a library of higher-order functions I think it's understood that every function is going to have another function as an argument.

For example, I have to read `!rejectionApplier(val)` very carefully.  Does `rejectionApplier` already do the negation?  Is this like double-not `!!val`?

Whereas something like:

```javascript
reject: (arr, func) => this.filter(arr, val => !func(val))
```

has almost no chance of being misunderstood at a glance.